promql/quantile.go

```diff
@@ -107,6 +109,51 @@ func bucketQuantile(q model.SampleValue, buckets buckets) float64 {
 	return bucketStart + (bucketEnd-bucketStart)*float64(rank/count)
 }
 
+// The assumption that bucket counts increase monotonically with increasing
+// upperBound may be violated during:
+//
+//   * Recording rule evaluation of histogram_quantile, especially when rate()
+//      has been applied to the underlying bucket timeseries.
+//   * Evaluation of histogram_quantile computed over federated bucket
+//      timeseries, especially when rate() has been applied
+//
+// This is because scraped data is not made available to RR evalution or
```

Comment:
"...available to rule evaluation..." reads nicer. (RR is not really a commonly used abbreviation. Besides, alerting rules are suffering from the same problem.)